### ISSUE TITLE
Move larger binary files to LFS

### ISSUE THREAD

**Issue by martukas** [RespiraWorks/Ventilator#863]
**What has to be done**
We have to
* enable git LFS for our repo
* transfer all the larger files (PCB stuff, in particular) to LFS
* scrub repo of the large files so they are not in repo proper
* document this **thoroughly** and make it easy for non-software-nerds to use, i.e. electrical and hardware engineers. This last point is super important!

**How do you know it has to be done**
Repo is taking too long to clone. It is now nearing 2GB.
We already knew this was going to be a problem before Covent.
Curtis has negotiated a free LFS package from github for us as a non-profit, so we should take advantage of it.

**Code pointers**
https://git-lfs.github.com/
https://docs.github.com/en/github/managing-large-files/configuring-git-large-file-storage

**Don't forget**
- [x] Put the issue in a milestone
- [x] Assign it to one of the projects
- [x] Add appropriate labels
- [x] Mention dependent/blocked issues if any

**Comment by martukas** [comment]
What is done:
* Most PCB and CAD files have been migrated.
* Entire repo history has been rewritten to reflect that.
* The github wiki pages has been updated with instructions for how to work with LFS.
* Cloning appears to be faster, and it will be also possible to clone without LFS. Let's see if CI speeds up.

What remains:
* The non-LFS part of the repo is still about 700MB
* It appears there are some JPEGs upwards of 5MB, and given that we have a lot of hardware assembly instructions and renderings, they might be adding up. Might need to do another pass.

The meh:
* Rewriting git history has invalidated some PRs, but nothing is lost that cannot be fixed with an old-fashioned rebase.

**Comment by martukas** [comment]
This is a list of issues and PRs were force-closed during the migration. Checklist to mark resolved. Resolved can mean
* if issue, repeopened
* if PR, branch rebased and new PR made

- [ ] #845 
- [ ] #769
- [ ] #763
- [ ] #515
- [ ] #185
- [ ] #439
- [ ] #117

If PR has to be reopened, please tag the old PR in the new one.

**Comment by martukas** [comment]
The procedure for migrating repo to LFS entails first fetching and pulling all branches from remote with
```
git branch -r | grep -v '\->' | while read remote; do git branch --track "${remote#origin/}" "$remote"; done
git fetch --all
git pull --all
```
Then if, for example, as I am about to do, you want to move all **jpeg** files to LFS you would:
```
git lfs migrate import --everything --include="*.jpeg,*.jpg"
```
This will rewrite the repo history from the beginning, and now you will have to do a force push:
```
git push -f
```
Remember to unblock master from being committed to and also enable force push on master in the repo settings.

**Comment by martukas** [comment]
After moving JPEGs now total non-LFS repo is about 800MB. Still hefty. Looks like the most likely culprit is data files, some of which ware 6MB. They are text files, so they do not belong in LFS. However, they are not really anything that deserved versioning, just archiving. They should likely not be text files at all, but in fact binary files, which would be orders of magnitude smaller. Likely candidate format would be something like the self-descriptive hdf5 (hierarchical data format) used by science facilities.

**Comment by martukas** [comment]
Two chores remain:
* configure CI to clone without LFS
* clean up PCB directory

**Comment by martukas** [comment]
Answering my own questions.
* CI does not need any changes. It clones only the pertinent branch and not the whole repo, which is fast anyways
* Cleaning up PCB directory is a separate ticket #873

**Comment by martukas** [comment]
Looks like `png` files might have to also be migrated to LFS.

**Comment by martukas** [comment]
On this day, performed one more migration for the following binary types:
* `png`
* `adicht` 
* `stl`
* `dxf`

**Comment by martukas** [comment]
On this day, adding *.idw

**Comment by martukas** [comment]
Additional casualties of this round:
#948 
#978 
#936 
#945
#904
#899
#973
